### PR TITLE
Add Zadig 101 to docs

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

A brief primer on how to use Zadig, how not to use it, and how to get yourself out of the "installed driver on wrong device" pickle.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/driver_installation_zadig.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+## Installation
+
+Place your keyboard into the bootloader mode, either by hitting the `RESET` keycode (which may be on a different layer), or by pressing the reset switch usually located on the underside of the board. If your keyboard has neither, try holding Escape, or Space+`B`, as you plug it in (see the [Bootmagic](feature_bootmagic.md) docs for more details).
+
+Zadig will automatically detect the bootloader device:
+ - For keyboards with Atmel AVR MCUs, the bootloader will be named something similar to `ATm32U4DFU`.
+ - AVR keyboards flashed with the QMK-DFU bootloader will be named `<keyboard name> Bootloader`.
+ - For most ARM keyboards, it will be called `STM32 BOOTLOADER`.
+
```
Comment (yiancar) [inline docs/driver_installation_zadig.md]: AVR Keyboards with USBaspbootloader name is `USBasp`

Comment (fauxpark) [reply docs/driver_installation_zadig.md]: Do they need one of the drivers provided by Zadig?

Comment (yiancar) [reply docs/driver_installation_zadig.md]: Yes usually libusb-win32 but some people said that didnt work and they had to use libusbk

Comment (fauxpark) [reply docs/driver_installation_zadig.md]: Done

**Comment by yiancar** [comment]
Thanks!

**Comment by drashna** [comment]
Don't forget to add a link to the _summary.md file.

**Comment by fauxpark** [comment]
Which section should it go in?

**Comment by drashna** [comment]
FAQ, probably.  That or maybe detailed guides.

File: docs/driver_installation_zadig.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Bootloader Driver Installation with Zadig
+
+QMK presents itself to the host as a regular HID keyboard device, and as such requires no special drivers. However, in order to flash your keyboard on Windows, the bootloader device that appears when you reset the board often *does*. There is one notable exception: the Caterina bootloader, usually seen on Pro Micros, appears as a serial port instead, and does not require a driver.
+
+We recommend the use of the [Zadig](https://zadig.akeo.ie/) utility. If you have set up the development environment with Msys2 or WSL, the `qmk_install.sh` script will have asked if you want it to install the drivers for you.
+
+## Installation
+
+Place your keyboard into the bootloader mode, either by hitting the `RESET` keycode (which may be on a different layer), or by pressing the reset switch usually located on the underside of the board. If your keyboard has neither, try holding Escape, or Space+`B`, as you plug it in (see the [Bootmagic](feature_bootmagic.md) docs for more details).
```
Comment (zvecr) [inline docs/driver_installation_zadig.md]: Might also be worth mentioning Bootmagic lite, and if the keyboards readme has any specific instructions.

**Comment by zvecr** [comment]
I would also note the docs I have been recommending recently are the cannonkeys ones <https://docs.cannonkeys.com/initial-flash-072/#one-time-setup>, while a little more geared towards one board, its more visual.

File: docs/driver_installation_zadig.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+Zadig will automatically detect the bootloader device:
+ - For keyboards with Atmel AVR MCUs, the bootloader will be named something similar to `ATm32U4DFU`.
+ - USBasp bootloaders will appear as `USBasp`.
+ - AVR keyboards flashed with the QMK-DFU bootloader will be named `<keyboard name> Bootloader`.
+ - For most ARM keyboards, it will be called `STM32 BOOTLOADER`.
+
+!> If Zadig lists one or more devices with the `HidUsb` driver, your keyboard is probably not in bootloader mode. The arrow will be colored orange and you will be asked to confirm modifying a system driver. **Do not** proceed if this is the case!
+
```
Comment (zvecr) [inline docs/driver_installation_zadig.md]: Some users have reported that "List All Devices" is required to see the bootloader device.

Comment (fauxpark) [reply docs/driver_installation_zadig.md]: Done

File: docs/driver_installation_zadig.md (hunk 2 above)
Comment (fauxpark) [reply docs/driver_installation_zadig.md]: Done

**Comment by zvecr** [comment]
We should also rip out most of https://docs.qmk.fm/#/faq_build?id=unknown-device-for-dfu-bootloader and point it this new central location.

**Comment by drashna** [comment]
Emphatically agree with @zvecr

**Comment by fauxpark** [comment]
How's this?

I've never actually seen the Toolbox ask to install drivers - and the occasional crash reports related to not having them installed suggests to me that it either doesn't work or is disabled in some way.

**Comment by drashna** [comment]
That looks good!

As for the drivers, it's in the readme, but it's .... not great:
https://github.com/qmk/qmk_toolbox#dependencies

**Comment by fauxpark** [comment]
(a version of) qmk_driver_installer.exe seems to be bundled with the Windows Toolbox, and I believe it's supposed to ask on first run, but I know for a fact that deleting the .config file from AppData doesn't make any difference.

**Comment by drashna** [comment]
Yeah, I think that's a big/issue for QMK Toolbox, but I have no idea how to fix that.  Though... it may be possible to script it in the installer?

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**